torchgeo/datasets/everwatch.py

```diff
@@ -0,0 +1,291 @@
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            AssertionError: If *split* argument is invalid.
+        """
+        assert split in self.valid_splits, (
+            f"Split '{split}' not supported, please use one of {self.valid_splits}"
+        )
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+        self.download = download
+
+        self._verify()
+
+        self.annot_df = pd.read_csv(
+            os.path.join(self.root, self.dir, f'{self.split}.csv')
+        )
+
+        # remove all entries where xmin == xmax or ymin == ymax
+        self.annot_df = self.annot_df[
+            (self.annot_df['xmin'] != self.annot_df['xmax'])
+            & (self.annot_df['ymin'] != self.annot_df['ymax'])
+        ].reset_index(drop=True)
+
+        # group per image path to get all annotations for one sample
+        self.annot_df['sample_index'] = pd.factorize(self.annot_df['image_path'])[0]
+        self.annot_df = self.annot_df.set_index(['sample_index', self.annot_df.index])
+
+        self.class2idx: dict[str, int] = {c: i for i, c in enumerate(self.classes)}
+
+    def __len__(self) -> int:
+        """Return the number of samples in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.annot_df.index.levels[0])
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        sample_df = self.annot_df.loc[index]
+
+        img_path = os.path.join(self.root, self.dir, sample_df['image_path'].iloc[0])
+
+        image = self._load_image(img_path)
+
+        boxes, labels = self._load_target(sample_df)
+
+        sample = {'image': image, 'bbox_xyxy': boxes, 'label': labels}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        with Image.open(path) as img:
+            array: np.typing.NDArray[np.uint8] = np.array(img)
+            tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, sample_df: pd.DataFrame) -> tuple[Tensor, Tensor]:
+        """Load target from a dataframe row.
+
+        Args:
+            sample_df: df subset with annotations for specific image
+
+        Returns:
+            bounding boxes and labels
+        """
+        boxes = torch.Tensor(
+            sample_df[['xmin', 'ymin', 'xmax', 'ymax']].values.tolist()
+        ).float()
```

Comment:
If I understand this line of code correctly, you convert from a pd.Series to a np.ndarray to a list to a torch.Tensor. I wonder if we can make this shorter, maybe skip the list?